lib/osc.dart

```diff
@@ -29,26 +29,21 @@ class Conn {
 
   Future<Message> receive(Duration timeout) async {
     // Create a disposable receiver using the same port as the sender's port
-    final receiver = await UDP.bind(Endpoint.any(port: sender.local.port));
-    var msg = Message();
     try {
-      await for (final event in receiver.asStream(timeout: timeout)) {
+      await for (final event in _sender.asStream(timeout: timeout)) {
         var data = event?.data ?? Uint8List(0);
-        if (data.isEmpty) throw Exception("empty packet");
-        msg = Message.fromPacket(data);
-        break; // once data is received, break out of the loop
+
+        print(String.fromCharCodes(data));
+        return Message.fromPacket(data);
       }
```

Comment:
Rather that using this for loop, and returning after the first event, ```Stream``` class has a property ```first``` which will return the first element and stop listening after.
```dart
datagram = await _sender.asStream().first;
```

Reply:
that is nice. let me try that.